forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/api/openbanking/payment/v3_1_4/domesticpayments/DomesticPaymentsApiController.java

```diff
@@ -25,25 +29,163 @@
+                .idempotentSave(frPaymentSubmission);
+        return ResponseEntity.status(HttpStatus.CREATED).body(packagePayment(frPaymentSubmission, paymentConsent));
+    }
+
+    @Override
+    public ResponseEntity getDomesticPaymentsDomesticPaymentId(
+            String domesticPaymentId,
+            String authorization,
+            DateTime xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            HttpServletRequest request,
+            Principal principal
+    ) throws OBErrorResponseException {
+        Optional<FRDomesticPaymentSubmission> isPaymentSubmission = domesticPaymentSubmissionRepository.findById(domesticPaymentId);
+        if (!isPaymentSubmission.isPresent()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Payment submission '" + domesticPaymentId + "' can't be found");
+        }
+        FRDomesticPaymentSubmission frPaymentSubmission = isPaymentSubmission.get();
+
+        Optional<FRDomesticConsent> isPaymentSetup = domesticConsentRepository.findById(domesticPaymentId);
+        if (!isPaymentSetup.isPresent()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Payment setup behind payment submission '" + domesticPaymentId + "' can't be found");
+        }
+        FRDomesticConsent frPaymentSetup = isPaymentSetup.get();
+        return ResponseEntity.ok(packagePayment(frPaymentSubmission, frPaymentSetup));
+    }
+
+    @Override
+    public ResponseEntity<OBWritePaymentDetailsResponse1> getDomesticPaymentsDomesticPaymentIdPaymentDetails(
+            String domesticPaymentId,
+            String authorization,
+            DateTime xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            HttpServletRequest request,
+            Principal principal
+    ) throws OBErrorResponseException {
+        // Optional endpoint - not implemented
+        return new ResponseEntity<OBWritePaymentDetailsResponse1>(HttpStatus.NOT_IMPLEMENTED);
+    }
+
+    private OBWriteDomesticResponse4 packagePayment(FRDomesticPaymentSubmission frPaymentSubmission, FRDomesticConsent frDomesticConsent) {
+        // flag set on the consent by the PISP if PSU requests a refund for the transaction.
+        // https://forgerock.zendesk.com/agent/tickets/55834
+        // https://github.com/OpenBankingToolkit/openbanking-toolkit/issues/14
+        boolean refund = frDomesticConsent.getDomesticConsent().getData().getReadRefundAccount().equals(FRReadRefundAccount.YES);
+
+        OBWriteDomestic2DataInitiation initiation = toOBWriteDomestic2DataInitiation(frDomesticConsent.getDomesticConsent().getData().getInitiation());
+
+        return new OBWriteDomesticResponse4()
+                .data(new OBWriteDomesticResponse4Data()
+                        .refund((refund && initiation.getDebtorAccount()!=null) ?
```

Comment:
I don't think this ternary that's over 9 lines is so easy to read. Perhaps extract this into a separate method? i.e. something like:

```
.data(new OBWriteDomesticResponse4Data()
   .refund(refundResponse(refund, initiation))
   .domesticPaymentId(frPaymentSubmission.getId())
```